assembly-syntax/src/ast/module.rs

```diff
@@ -322,6 +391,26 @@ impl Module {
         &self.advice_map
     }
 
+    /// Get an iterator over the constants defined in this module.
+    pub fn constants(&self) -> core::slice::Iter<'_, Constant> {
```

Comment:
nit: any reason to specify the current iterator used as opposed to `impl Iterator<...>`?

Applies to other methods too

Reply:
For slice iterators in particular, it allows for better optimizations due to all of the traits implemented by the concrete `Iterator` type, i.e. `DoubleEndedIterator`, `ExactSizeIterator`, `TrustedLen`, `FusedIterator`, etc. Additionally, it is possible to recover the underlying slice from it, so it is considerably more flexible.